### PR TITLE
XW-2904 | ArticleAsJson: check if 'caption' field is set

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/XW-2904

check if 'caption' field is set in ArticleAsJson::onImageBeforeProduceHtml

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 1]
```diff
@@ -306,7 +306,8 @@ public static function onImageBeforeProduceHTML(
 			$details['context'] = self::isIconImage( $details, $handlerParams ) ? self::MEDIA_CONTEXT_ICON :
 				self::MEDIA_CONTEXT_ARTICLE_IMAGE;
 
-			$media = self::createMediaObject( $details, $title->getText(), $frameParams['caption'], $linkHref );
+			$caption = isset( $frameParams['caption'] ) ? $frameParams['caption'] : null;
```
Comment (rogatty) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Would `$caption = $frameParams['caption'] ?? null` work?

**Review by rogatty — APPROVED** [review]

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php (hunk 1 above)
Comment (rybmat) [reply extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: ?? checks if variable is null, not sure how this will perform on missing key in array

Comment (rybmat) [reply extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: but I'll check

Comment (rybmat) [reply extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: ok, it should work

**Pull request merged**